### PR TITLE
[Feature] implement bitmap_subset_in_range function

### PR DESCRIPTION
**Pull request by ucasfl**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by Seaven** [comment]
soo....why don't implement `bitmap_subset_limit_internal` use `bitmap_subset_in_range_internal(0, limit, bitmap)`?

**Comment by ucasfl** [comment]
> soo....why don't implement `bitmap_subset_limit_internal` by call `bitmap_subset_in_range_internal(0, limit, bitmap)`?

`bitmap_subset_limit` have two other arguments except bitmap: start_range and limit, and limit can be negtive, so it's not equal to `bitmap_subset_in_rnage_internal(0, limit, bitmap)`.

**Comment by Seaven** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by Seaven** [comment]
run all

**Comment by ucasfl** [comment]
> run all

Unite test failed, I'm checking locally.

**Comment by ucasfl** [comment]
```
:52:03 tracker:consistency consumption: 0
14:52:03 *** Aborted at 1677826322 (unix time) try "date -d @1677826322" if you are using GNU date ***
14:52:03 PC: @          0xdd5dacf starrocks::ColumnViewer<>::value()
14:52:03 *** SIGSEGV (@0x0) received by PID 12528 (TID 0x7f9fcf098440) from PID 0; stack trace: ***
14:52:03     @         0x16787502 google::(anonymous namespace)::FailureSignalHandler()
14:52:03     @     0x7f9fcdc2b630 (unknown)
14:52:03     @          0xdd5dacf starrocks::ColumnViewer<>::value()
14:52:03     @         0x1188861a starrocks::BitmapFunctions::bitmap_subset_in_range()
14:52:03     @          0xb3f3533 starrocks::VecBitmapFunctionsTest_bitmap_subset_in_range_Test::TestBody()
14:52:03     @         0x18670791 testing::internal::HandleExceptionsInMethodIfSupported<>()
14:52:03     @         0x18667126 testing::Test::Run()
14:52:03     @         0x18667295 testing::TestInfo::Run()
14:52:03     @         0x18667385 testing::TestSuite::Run()
14:52:03     @         0x186678d6 testing::internal::UnitTestImpl::RunAllTests()
14:52:03     @         0x18667ade testing::UnitTest::Run()
14:52:03     @          0x9deb34c RUN_ALL_TESTS()
14:52:03     @          0x9de15ca main
14:52:03     @     0x7f9fcd162555 __libc_start_main
14:52:03     @          0x9d1a88f (unknown)
14:52:03     @                0x0 (unknown)
14:52:03 run-be-ut.sh: line 253: 12528 Segmentation fault      ${GTEST_PARALLEL} ${STARROCKS_TEST_BINARY_DIR}/starrocks_test --gtest_filter=${TEST_NAME} --serialize_test_cases ${GTEST_PARALLEL_OPTIONS}
14:52:03 + chown -R jenkins:jenkins /home/disk3/jenkins/workspace/starrocks_be_unittest/cluster-management/resource/starrocks/
14:52:03 + '[' -f be_ut_status ']'
14:52:03 ++ cat be_ut_status
14:52:03 + a=1
14:52:03 + '[' 1 == 1 ']'
14:52:03 + tar -czvf be-ut-log-18589.tar.gz log
14:52:03 log/
14:52:03 log/be_test.INFO.log.20220617-133629
14:52:03 log/be_test.INFO
14:52:08 log/be_test.WARNING.log.20220617-133630
14:52:08 log/be_test.WARNING
14:52:11 + sshpass -p dorisdb@test scp -o StrictHostKeyChecking=no be-ut-log-18589.tar.gz jenkins@ci01:/home/disk1/jenkins/userContent/
14:52:11 + exit 1
14:52:12 Build step 'Execute shell' marked build as failure
14:52:12 Finished: FAILURE
```
I have no idea why the core happened. @Seaven

**Comment by Seaven** [comment]
> ColumnViewer

don't worry，let me test it later

File: be/test/exprs/bitmap_functions_test.cpp [hunk 1]
```diff
@@ -2298,4 +2298,215 @@ TEST_F(VecBitmapFunctionsTest, sub_bitmap) {
...
+        auto column = BitmapFunctions::bitmap_subset_in_range(ctx, columns).value();
+        ASSERT_TRUE(column->is_null(0));
+    }
+
+    {
+        Columns columns;
+        auto start = Int64Column::create();
+        auto end = Int64Column::create();
+        start->append(0);
+        start->append(-5);
```
Comment (Seaven) [inline be/test/exprs/bitmap_functions_test.cpp:2469]: I think the crash reason is don't set `end` column

**Comment by ucasfl** [comment]
Failed FE unittest seems unrelated.

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_clang-format

**Review by Seaven — DISMISSED** [review]

File: be/src/exprs/bitmap_functions.cpp [hunk 2]
```diff
@@ -569,4 +569,47 @@ StatusOr<ColumnPtr> BitmapFunctions::bitmap_to_base64(FunctionContext* context,
...
+        auto bitmap = bitmap_viewer.value(row);
+        auto range_start = range_start_viewer.value(row);
+        auto range_end = range_end_viewer.value(row);
+
+        if (range_start < 0) {
+            range_start = 0;
+        }
+
+        if (bitmap->cardinality() == 0 || range_start >= range_end) {
+            builder.append_null();
```
Comment (fzhedu) [inline be/src/exprs/bitmap_functions.cpp:599]: empty bitmap returns null by design?

**Review by fzhedu — COMMENTED** [review]
pls add doc

File: be/src/exprs/bitmap_functions.cpp (hunk 2 above)
Comment (ucasfl) [reply be/src/exprs/bitmap_functions.cpp:599]: Yes.

**Review by Seaven — DISMISSED** [review]
LGTM, wait the docment enginer review plz

**Review by fzhedu — DISMISSED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Seaven** [comment]
run all

**Comment by wangruin** [comment]
run all

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by ucasfl** [comment]
Build failed, unrelated.
```
[ 81%] Building CXX object test/CMakeFiles/starrocks_test.dir/exprs/percentile_functions_test.cpp.o
16:21:55 /root/starrocks/be/test/exprs/array_functions_test.cpp: In member function 'virtual void starrocks::ArrayFunctionsTest_array_intersect_any_type_int_with_not_null_Test::TestBody()':
16:21:55 /root/starrocks/be/test/exprs/array_functions_test.cpp:5032:43: error: unused variable 'intersect' [-Werror=unused-variable]
16:21:55  5032 |     ArrayIntersect<LogicalType::TYPE_INT> intersect;
16:21:55       |                                           ^~~~~~~~~
16:21:55 cc1plus: all warnings being treated as errors
16:21:55 make[2]: *** [test/CMakeFiles/starrocks_test.dir/exprs/array_functions_test.cpp.o] Error 1
16:21:55 make[2]: *** Waiting for unfinished jobs....
16:22:15 make[1]: *** [test/CMakeFiles/starrocks_test.dir/all] Error 2
16:22:15 make[1]: *** Waiting for unfinished jobs....
16:26:22 [ 81%] Built target engine_storage_migration_task_test
16:26:22 make: *** [all] Error 2
16:26:22 + chown -R jenkins:jenkins /home/disk4/jenkins/workspace/starrocks_be_unittest/cluster-management/resource/starrocks/
16:26:22 + '[' -f be_ut_status ']'
16:26:22 ++ cat be_ut_status
16:26:22 + a=1
16:26:22 + '[' 1 == 1 ']'
16:26:22 + tar -czvf be-ut-log-18589.tar.gz log
16:26:22 log/
16:26:22 log/be_test.WARNING
16:26:22 log/be_test.INFO.log.20220617-133629
16:26:31 log/be_test.WARNING.log.20220617-133630
16:26:34 log/be_test.INFO
16:26:34 + sshpass -p dorisdb@test scp -o StrictHostKeyChecking=no be-ut-log-18589.tar.gz jenkins@ci01:/home/disk1/jenkins/userContent/
16:26:34 + exit 1
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Seaven — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**